### PR TITLE
executor: optimize delete statement.

### PR DESCRIPTION
**Pull request by coocood**
Do not call table.Row() for every deleted row.

The time of deleting 100,000 rows from a table has reduced from 18.3s to 2.3s.

### CODE REVIEW

File: executor/executor_write.go [hunk 1]
```diff
@@ -166,53 +166,58 @@ func (e *DeleteExec) Next() (*Row, error) {
...
 
 	tblMap := make(map[int64][]string, len(e.Tables))
 	// Get table alias map.
-	if e.IsMultiTable {
-		// Delete from multiple tables should consider table ident list.
-		for _, t := range e.Tables {
-			tblMap[t.TableInfo.ID] = append(tblMap[t.TableInfo.ID], t.Name.L)
-		}
+
+	// Delete from multiple tables should consider table ident list.
```
Comment (hanfei1991) [inline executor/executor_write.go]: Put this comment together with line 182?

File: executor/executor_write.go [hunk 2]
```diff
@@ -166,53 +166,58 @@ func (e *DeleteExec) Next() (*Row, error) {
...
+	}
+	return nil, e.deleteSingleTable()
+}
+
+func (e *DeleteExec) deleteMultiTables() error {
+	if len(e.Tables) == 0 {
+		return nil
 	}
 
 	tblMap := make(map[int64][]string, len(e.Tables))
```
Comment (hanfei1991) [inline executor/executor_write.go]: add comment to explain the meaning of map key.

File: executor/executor_write.go [hunk 3]
```diff
@@ -166,53 +166,58 @@ func (e *DeleteExec) Next() (*Row, error) {
...
 	// Get table alias map.
-	if e.IsMultiTable {
-		// Delete from multiple tables should consider table ident list.
-		for _, t := range e.Tables {
-			tblMap[t.TableInfo.ID] = append(tblMap[t.TableInfo.ID], t.Name.L)
-		}
+
+	// Delete from multiple tables should consider table ident list.
+	for _, t := range e.Tables {
+		tblMap[t.TableInfo.ID] = append(tblMap[t.TableInfo.ID], t.Name.L)
```
Comment (hanfei1991) [inline executor/executor_write.go]: When two different names stand for the same table?

Comment (coocood) [inline executor/executor_write.go]: `delete from t as t1, t as t2`

**Comment by hanfei1991** [comment]
LGTM

File: executor/explain_test.go [hunk 4]
```diff
@@ -260,7 +260,7 @@ func (s *testSuite) TestExplain(c *C) {
     "ranges": "[[1,1]]",
     "desc": false,
     "out of order": true,
-    "double read": false,
+    "double read": true,
```
Comment (tiancaiamao) [inline executor/explain_test.go]: what does this mean?

Comment (coocood) [inline executor/explain_test.go]: double read first read handle from index request, then use handle to send table request to read full row.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**